src/clj/puppetlabs/services/legacy_routes/legacy_routes_core.clj

```diff
@@ -0,0 +1,144 @@
+(ns puppetlabs.services.legacy-routes.legacy-routes-core
+  (:require [clojure.string :as str]
+            [clojure.tools.logging :as log]
+            [compojure.core :as compojure]
+            [ring.util.codec :as ring-codec]
+            [compojure.route :as route]
+            [clojure.tools.logging :as log]))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; v3 => v4 header translation support functions
+
+; TODO: Improve the naive parsing here with a real library like
+; https://github.com/ToBeReplaced/http-accept-headers
```

Comment:
As a general practice, copying functionality from a non-PL library into a PL one rather than referencing the project directly as a dependency doesn't quite feel right to me.  Granted, there's a point at which if you're just talking about one small utility function which is unrelated to the overall purpose of the library it is coming from where this could make sense and this might be a little closer to that camp.

Ultimately, we need functionality here that would not only parse out the values for the header but be able to flatten them back into a valid string after we apply some changes.  It wasn't clear to me that the liberator implementation provided the latter.  Maybe the `ToBeReplaced/http-accept-headers` library does and could, therefore, be a better fit for our use case anyway - especially if we're still hoping to sever all ties with liberator from Puppet Server.

In any event, I think the simpler implementation in this PR that doesn't account for quality weights is probably sufficient for now as it should cover how actual agents populate the header.  So I'd be good with this implementation coming in as-is for now and maybe just file a separate ticket to explore pulling in a more complete upstream library later on.  I wouldn't block the next release for that latter work being done.